packages/asset-service/src/generateAssetData/setColors.ts

```diff
@@ -0,0 +1,27 @@
+import { Asset } from '@shapeshiftoss/types'
+import colorThief from 'colorthief'
+
+function toHex(num: number): string {
+  const hex = num.toString(16).toUpperCase()
+  return hex.length === 1 ? `0${hex}` : hex
+}
+
+export const setColors = async (assets: Asset[]): Promise<Asset[]> => {
+  for (let i = 0; i < assets.length - 1; i++) {
+    try {
+      if (assets[i].color === '#FFFFFF' && assets[i].icon) {
+        // colorThief.getColor returns the most dominant color in the icon.
+        const color = await colorThief.getColor(assets[i].icon)
+        const hexColor = `#${toHex(color[0])}${toHex(color[1])}${toHex(color[2])}`
```

Comment:
Non-blocking, I'd name these `red` `green` and `blue` (or `r`, `g` and `b`) but given we're probably not going to change that script's logic, I'm fine with this

Reply:
Good call. Destructuring makes sense.